core/engine/src/value/conversions/try_from_js.rs

```diff
@@ -156,6 +156,47 @@ impl TryFromJs for f64 {
     }
 }
 
+trait ToF64: Sized {
```

Comment:
thought: Is there any gain from manually implementing this vs using `NumTraits`' `ToPrimitive`/`FromPrimitive`?

Reply:
I started using the `ToPrimitive`/`FromPrimitive` but found out that it's not lossless :( https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a15242f1b391f3de4ba30314190e4add

Reply:
I'm not following why we need to care about the floating point case when we're trying to convert from an integer f64 to an integer

Reply:
Huh, that's true. Then `NumTraits` definitely isnt needed

Reply:
I think we're slightly talking over each other. I'm talking about still having the check for equality with the bits representation of the original number, but just using `To/FromPrimitive` to do the conversion to integer instead of a new trait. Hence:

```rust
fn from_f64<T: ToPrimitive + FromPrimitive>(v: f64) -> Option<T> {
    let i = T::from_f64(v)?;
    if i.to_f64()?.to_bits() != v.to_bits() {
        return None;
    }
    Ok(i)
}
```

Reply:
I see, yeah we can use it, `AsPrimitive` seems perfect for the job, thank for the suggestion :)